# URM

In [3]:
from Utils.Toolkit import get_data
import numpy as np

urm = get_data()['URM_all']
urm = urm.tocsr()
urm

<30911x18495 sparse matrix of type '<class 'numpy.float64'>'
	with 398636 stored elements in Compressed Sparse Row format>

# Cold Users

In [41]:
cold_mask = np.ediff1d(urm.indptr) == 0
user_ids = np.arange(urm.shape[0])
cold_user_ids = user_ids[cold_mask]
len(cold_user_ids)

3656

# Target Users

In [42]:
target_user_ids = get_data()['target_users']
len(target_user_ids)

28106

#### Each cold user in the URM is also in the target users

In [43]:
np.in1d(cold_user_ids, target_user_ids).all()

True

So there are 3656 users on which we won't be albe to test the correctness of the recommendation

In [44]:
print(len(target_user_ids)-len(cold_user_ids))

24450


There are 24450 users on which we can calculate the MAP.

How many of them will be cold after train/test split?

In [45]:
cold_mask = np.ediff1d(urm.indptr) == 1
user_ids = np.arange(urm.shape[0])
cold_user_ids = user_ids[cold_mask]
len(cold_user_ids)

2817

In [46]:
print(f'{len(target_user_ids)-len(cold_user_ids)-len(cold_user_ids)} warm users')

22472 warm users


# Recap

In [47]:
print(f'Num of users: {urm.shape[0]}')
print(f'Num of items: {urm.shape[1]}')
print(f'Num of cold users in URM on which we cannot calculate how good our recommender is 3656')
print(f'Num of target users is ')

Num of users: 30911
Num of items: 18495
Num of cold users in URM on which we cannot calculate how good our recommender is 3656
